### PR TITLE
monitoring: wrap cadvisor metrics in label_replace

### PR DESCRIPTION
**Pull request by bobheadxi**
Closes https://github.com/sourcegraph/sourcegraph/issues/11571

It seems the `name` label has some special undocumented purpose, which was causing the [provisioning `alert_rules` to fail to create](https://github.com/sourcegraph/sourcegraph/issues/11571#issuecomment-645921031).

This PR wraps each of the cAdvisor dashboards with a `label_replace` to add a `container` label that is just the value of the `name` label. I manually updated Prometheus on k8s.sgdev.org and it seems to do the trick for alert_counts that were previously failing:

Screen Shot 2020-07-02 at 5 25 37 PM

This PR also includes two misc. improvements:

* remove `(not available on server)` from individual metrics - this text already shows up on dashboard section titles
* fix formatting for possible solutions

### CODE REVIEW

**Comment by bobheadxi** [comment]
Just to get this down in writing, manual testing involved:

1. generate new alert rules locally
2. ssh into prometheus container in k8s.sgdev.org
3. replace one of the alert_rules files
4. `kill -HUP $(pgrep prometheus)` to restart prometheus

**Comment by bobheadxi** [comment]
tl;dr - the `name` label alone seems insufficient to have Prometheus treat the cAdvisor series as unique (?)

Why `label_replace`? I figured this might work, and this was the only way to generate that label that I can take and POC on the current problematic deployments (dot-com, k8s) without spinning up a deployment myself. It might not be the best way to do this, however:

* have cAdvisor export a different label - AFAIK, there is no configuration option for this
* do the label replacement in `cadvisor_rules.yml` and `cadvisor_container_restart_count_rules.yml`. On paper this is pretty close to the same thing - I plan to investigate this tomorrow to see if it has any advantage over the current approach

**Comment by bobheadxi** [comment]
6588ad7 moves the `label_replace` to our aggregated cAdvisor rules instead of having them on individual dashboards - it's taking a while, but it seems the provisioning rules are applying more consistently now (k8s):

image

I tried using `label` templates, but I don't believe recording rules support them: https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/#rule

**Comment by slimsag** [comment]
I think I found the actual underlying cause and have suggested an alternative fix here: https://github.com/sourcegraph/sourcegraph/issues/11571#issuecomment-654571953
